src/lib/mina_base/proof_cache_tag/lmdb/proof_cache_tag.ml

```diff
@@ -0,0 +1,34 @@
+(* Cache proofs using the lmdb *)
+
+open Core
+open Lmdb_storage.Generic
+
+type value = Pickles.Proof.Proofs_verified_2.Stable.Latest.t
+
+module F (Db : Db) = struct
+  type holder = (int, value) Db.t
+
+  let mk_maps { Db.create } =
+    create Lmdb_storage.Conv.uint32_be
+      (Lmdb_storage.Conv.bin_prot_conv
+         Pickles.Proof.Proofs_verified_2.Stable.Latest.bin_t )
+
+  let config = { default_config with initial_mmap_size = 256 lsl 20 }
+end
+
+module Rw = Read_write (F)
+
+let db_dir = Filename.temp_dir "mina" "proof_cache.lmdb"
+
+type t = { idx : int } [@@deriving compare, equal, sexp, yojson, hash]
+
+let unwrap ({ idx = x } : t) : value =
+  (* Read from the db. *)
+  let env, db = Rw.create db_dir in
+  Rw.get ~env db x |> Option.value_exn
+
+let generate (x : value) : t =
```

Comment:
In the current implementation DB is only expanded and never shrinked

I propose to use implementation similar to `proof_cache_tag.filesystem`:

```ocaml
let idx= !counter in
  incr counter ;
  let res = { idx } in
  (* When this reference is GC'd, delete the file. *)
  Gc.Expert.add_finalizer_last_exn res (fun () ->
  (* TODO remove the `idx` key from DB *)
       ) ;
```
Note that hashing is dropped in favor of indexing. While it may result in higher RAM usage, we don't need to care about hash collisions (`hash` is a non-cryptographic hash), and the logic to free up disk becomes as simple as above.